src/pip/_internal/vcs/subversion.py

```diff
@@ -33,6 +40,22 @@ def should_add_vcs_url_prefix(cls, remote_url):
     def get_base_rev_args(rev):
         return ['-r', rev]
 
+    def get_vcs_version(self):
+        # type: (...) -> Union[LegacyVersion, Version]
+        """Gets the version of the currently installed Subversion client."""
+        VERSION_PFX = 'svn, version '
+        version = self.run_command(['--version'], show_stdout=False)
+        if version.startswith(VERSION_PFX):
+            version = version[len(VERSION_PFX):].split()[0]
+        else:
+            version = ''
+
+        # Example versions:
+        #   svn, version 1.10.3 (r1842928)
+        #   svn, version 1.7.14 (r1542130)
+        version = '.'.join(version.split('.')[:3])
+        return parse_version(version)
```

Comment:
Since this isn't the version of a Python package, I think it would be better / simpler just to return a tuple (a bit like `sys.version_info`).

Reply:
@cjerdonek In this case I am trying to be consistent with `Git.get_git_version`. I think in the future, `pip` may want to create an abstract function `VersionControl` with the same type signature for all child classes. I think that's out of scope for this PR, but at least this keeps `Subversion` working more closely to how `Git` does.

See https://github.com/pypa/pip/pull/6390#issuecomment-482804947

Reply:
> I think in the future, pip may want to create an abstract function VersionControl with the same type signature for all child classes.

Right. The suggestions I'm giving you are how we'd want that abstract function to be. Otherwise, two functions would have to be fixed instead of one. So it's better to do the SVN one right the first time.